src/jdk.internal.le/linux/native/lible/CLibrary.cpp

```diff
@@ -188,6 +188,7 @@ JNIEXPORT void JNICALL Java_jdk_internal_org_jline_terminal_impl_jna_linux_CLibr
 
     if (error != 0) {
         throw_errno(env);
+        delete[] data;
```

Comment:
While JNI would not transfer control immediately on `Throw`, and would only throw it on return, it is a good style to do cleanups first, and then do throw, followed by immediate return.

In other words, flip the order of `delete[]` and `throw_errno`.

Reply:
oh! got it